typescript/infra/scripts/funding/print-dollar-value-key-funder-threshold.ts

```diff
@@ -0,0 +1,109 @@
+import lowUrgencyKeyFunderBalance from '../../config/environments/mainnet3/balances/lowUrgencyKeyFunderBalance.json' with { type: 'json' };
+import targetDollarBalances from '../../config/environments/mainnet3/balances/target-dollar-balances.json' with { type: 'json' };
+import { mainnet3SupportedChainNames } from '../../config/environments/mainnet3/supportedChainNames.js';
+import tokenPrices from '../../config/environments/mainnet3/tokenPrices.json' with { type: 'json' };
+
+type ChainName = keyof typeof targetDollarBalances;
+
+// Get chains from targetDollarBalances
+const CHAINS = Object.keys(targetDollarBalances) as ChainName[];
+
+// Validate that all CHAINS are in mainnet3SupportedChainNames
+const unsupportedChains = CHAINS.filter(
+  (chain) => !mainnet3SupportedChainNames.includes(chain),
+);
+if (unsupportedChains.length > 0) {
+  console.error(
+    `Error: The following chains are not in mainnet3SupportedChainNames:\n${unsupportedChains.join('\n')}`,
+  );
+  process.exit(1);
+}
+
+interface ChainBalanceInfo {
+  chain: ChainName;
+  symbol: string;
+  tokenPrice: number;
+  balance: number;
+  dollarValue: number;
+  targetDollarValue: number;
+  percentageOfTarget: number;
+  excessFunding: number;
+}
+
+function processChains(
+  chains: readonly ChainName[],
+  balances: Record<ChainName, number>,
```

Comment:
nit: this is referred to as "balances" but it's just the thresholds not the balance itself